.github/workflows/typedoc.yml

```diff
@@ -0,0 +1,70 @@
+name: Typedoc
+
+on:
+  push:
+    branches:
+      - master  # Set a branch name to trigger deployment
+      - typedoc-action
+
+jobs:
+  deploy:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true  # Fetch Hugo themes (true OR recursive)
+          fetch-depth: 0    # Fetch all history for .GitInfo and .Lastmod
+
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+
+      - name: Install
+        env:
+          RUNNING_NODE_CI: 1
+        run: npm ci
+
+      - name: Build
+        run: npm run build
+
+      - name: Generate
+        run: npx lerna run doc:generate
+
+      - name: Deploy - msal-core
```

Comment:
If these deploy steps are all the same but for different libs you could use a matrix strategy instead. You can take a look at our CI steps to see how this is done [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/.github/workflows/build-steps.yml#L23)

The benefit is that each one will run in parallel and if one fails it won't stop deployment of the others

Reply:
Good to know, I'll look at doing this incrementally.